src/nvim/eval.lua

```diff
@@ -20,6 +20,8 @@
 --- @field signature? string
 --- @field desc? string
 --- @field params [string, string, string][]
+--- @field notes? string[]
+--- @field see? string[]
```

Comment:
Currently only used by gen_eval_files, but we probably want this structure for eval.lua in the future?